feat: xblock asides view for rendering in an iframe in MFE

Anas12091101 opened the pull request:
<!--

🌳🌳
🌳🌳🌳🌳         🌳 Note: Quince is in support. Fixes you make on master may still be needed on Quince.
    🌳🌳🌳🌳     If so, make another pull request against the open-release/quince.master branch,
🌳🌳🌳🌳         or ask in the #wg-build-test-release Slack channel if you have any questions or need help.
🌳🌳


Please give your pull request a short but descriptive title.
Use conventional commits to separate and summarize commits logically:
https://open-edx-proposals.readthedocs.io/en/latest/oep-0051-bp-conventional-commits.html

Use this template as a guide. Omit sections that don't apply.
You may link to information rather than copy it, but only if the link is publicly
readable.  If you must linked information must be private (because it has secrets),
clearly label the link as private.

-->

## Description

This PR introduces an aside only studio view in edx-platform for any Xblock i.e, given an Xblock usage key, get all the asides and render them into an HTTP response template page. This can then be utilized into an iframe in the course authoring MFE which currently doesn't support any asides configuration.

Useful information to include:

- The change in this PR will impact the course author using course-authoring MFE.
- Screenshots:
<img width="941" alt="Screenshot 2024-01-05 at 3 57 27 PM" src="https://github.com/Anas12091101/edx-platform/assets/88967643/702fbae1-ce10-4b87-b182-321936fdab82">
<img width="941" alt="Screenshot 2024-01-05 at 3 59 47 PM" src="https://github.com/Anas12091101/edx-platform/assets/88967643/c22f0547-115c-4ae2-87f6-189bba9a6c03">

## Supporting information

Related [Issue](https://github.com/openedx/frontend-app-course-authoring/issues/545)

## Testing instructions

Please provide detailed step-by-step instructions for testing this change.
- Login to edx-platform.
- In Studio go to the edX Demo Course. Create a new unit which is a multiple choice problem.
- Confgure rapid response aside for this problem from the steps mentioned in it's [repository](https://github.com/mitodl/rapid-response-xblock).
- Go to the URL http://localhost:18010/asides/{xblock-usage-key-for-the-problem} and you will see the rapid response aside upon loading completion.

mariajgrimaldi committed:
feat: remove edX branding on account deletion

Agrendalath committed:
feat: allow disabling user tours

A similar waffle flag was dropped in 20386337.

OmarIthawi committed:
feat: atlas pull plugins translation

ormsbee committed:
docs: ADR for course data independence from libraries

This is an ADR around the idea that course content should not need to
sync metadata from libraries during the import process or when copy and
pasting.

ormsbee committed:
fix: make content_libraries ADR headers match doc guidelines

https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#headings

Commit:
feat: able to render aside only without xblock

Commit:
feat: asides only view working

Anas12091101 commented on cms/lib/xblock/tagging/tagging.py:
```diff
@@ -71,6 +71,48 @@ def student_view_aside(self, block, context):  # pylint: disable=unused-argument
             return fragment
         else:
             return Fragment('')
+        
```
Testing multiple studio_view asides

Anas12091101 commented on cms/lib/xblock/tagging/tagging.py:
```diff
@@ -71,6 +71,48 @@ def student_view_aside(self, block, context):  # pylint: disable=unused-argument
+            fragment.initialize_js('StructuredTagsInit')
+            return fragment
+        else:
+            return Fragment('')
```
Testing multiple asides

Commit:
feat: some cleaning

Commit:
feat: cleaned up unuseful things

Commit:
feat: xblock asides view for rendering in an iframe in MFE

pdpinch commented:
@Anas12091101 can you clarify this? I believe this view was functional in the old Studio UI, but it doesn't work in the course authoring MFE. So this is restoring existing functionality in the new MFE UI.

Anas12091101 commented:
@pdpinch yes, you are right. In the legacy studio we can access this view in the plugins tab which displays all the available plugins for the xblock installed in the edx-platform. But in the new `course authoring MFE`, there is no method to set the values of the plugins. The approach used in this PR is inspired by the one used by the `learning MFE` which loads the `student_view` along with the `asides` in an iframe. However there's one change. We don't need to load the legacy `studio_view` of the problem xblock as the course authoring is already parsing the problem and rendering it into it's own components. Hence we only need the xblock's `asides studio_view` which can easily be rendered in an iframe in course authoring MFE.

varshamenon4 committed:
refactor: move rejected exam event handlers

leangseu-edx committed:
chore: revert change to Mathjax

Commit:
temp: rebased with master

iamsobanjaved committed:
chore!: upgrade Django version to 4.2 (LTS)

This reverts commit 6f32d0e6f1596445822a7e964ffd6928d0d5415b.

Commit:
feat: some cleanup

mariajgrimaldi committed:
refactor: address PR reviews

rijuma committed:
fix: Updated log to have a cleaner output on failed reindexes

Agrendalath committed:
fix: handle countries not recognized by geoip2 DB

The `geoip2` library returns `None` when it only identifies the continent
related to the IP address.

macdiesel committed:
chore: Upgrade the jquery cookie library.

macdiesel committed:
Merge pull request #34109 from openedx/bbeggs/jquery.cookie

Chore: Upgrade the jquery cookie library.

macdiesel committed:
Revert "refactor: changed survey report message location and added a new info…"

macdiesel committed:
Merge pull request #34110 from openedx/revert-34033-ase/refactor-survey-report

Revert "refactor: changed survey report message location and added a new info…"

rijuma committed:
Merge pull request #34105 from openedx/rijuma/update-reindex_couses-output

Updated reindex_course console output to have cleaner data and a list of failing courses

github-actions[bot] committed:
feat: Upgrade Python dependency edx-drf-extensions (#34114)

make forgiving JWTs the default

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Co-authored-by: robrap <robrap@users.noreply.github.com>

Commit:
feat: bring in new client id configuration from ebk (#34115)

pomegranited committed:
feat: tagging: serialize object permissions to REST API (#34004)

ChrisChV committed:
feat: Tagging UX refinements - refresh tag count on edit (#34059)

* style: drawer-cover color updated for all tagging drawers
* feat: Update TagList component when a tag is updated on Manage tags drawer
* feat: Refactor TagCount to be able to refresh the count
* feat: Sync tag count in units

deborahgu committed:
feat: data migration for some ancient legacy problems

one time data migration for  certificates that ended up in a broken
state (not marked is unavailable, but having invalidation records).

FIXES: APER-1322

deborahgu committed:
Merge branch 'master' into dkaplan1/APER-1322_certs-with-invalidation-records-arent-in-the-unavailable-status

iamsobanjaved committed:
feat: Upgrade Python dependency Django

chore: upgrade Django to latest patch version

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

iamsobanjaved committed:
Merge pull request #34119 from openedx/iamsobanjaved/upgrade-Django-67d53dc

feat: Upgrade Python dependency Django

saadyousafarbi committed:
Revert "temp: add temporary logs for course wide notifications"

UsamaSadiq committed:
Pin optimizely sdk (#34121)

* fix: pin optimizely-sdk==4.1.1

* fix: update optmizely-sdk constraint

* feat: Upgrade Python dependency optimizely-sdk (#34120)

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Co-authored-by: UsamaSadiq <UsamaSadiq@users.noreply.github.com>

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: UsamaSadiq <UsamaSadiq@users.noreply.github.com>

github-actions[bot] committed:
feat: Upgrade Python dependency django-pipeline (#34124)

fix: upgrade django-pipeline package

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Co-authored-by: UsamaSadiq <UsamaSadiq@users.noreply.github.com>

edx-requirements-bot committed:
Python Requirements Update (#34123)

* chore: Updating Python Requirements
* fix: pin lxml<5.0
* chore: Updating Python Requirements (#34125)

---------

Co-authored-by: Usama Sadiq <usama.sadiq@arbisoft.com>

johnnagro committed:
feat: release edx-enterprise 4.10.10 (#34128)

Commit:
feat: add auth JWKS endpoint

Adds an endpoint to expose JWT_PUBLIC_SIGNING_JWK_SET to other services

deborahgu committed:
feat: fix exception handling in program cert revocation

To determine whether or not we need to  revoke any program certificates, we first run get_revokable_program_uuids. This calls get_certified_programs, which calls get_credentials, which uses the OpenEdx Core  utility method get_api_data. get_api_data makes the API call inside a try block, does raise_for_status  also inside the try block, and then catches the exception, logs it, and returns an empty result to the caller.

This means that on a failure to call the credentials API, get_credentials can’t  tell the difference between a failure to hit the API (because credentials is, as it sometimes is during a notify_programs run, overloaded), or a learner with no program certificates. In this particular case, this is absolute failure, incorrect behavior.

* Adds a new flag, `raise_on_error`  which will make `get_api_data` log the exception and then re-raise the HTTPError,  defaulting to false in order to avoid changing the behavior on any other callers
* Also: my editor reformatted all of the touched files to our modern code standards, and it seemed appropriate to let it do that.
* Also: added type hints in some cases, because they helped me write the code and debug. Our test suite definitely  reports mypy  results on type errors so we are verifying that hints are correct.

FIXES: APER-3146

Commit:
feat: Added XblockMixin for skill tagging

iloveagent57 committed:
fix: edx-enterprise 4.10.11

AhtishamShahid committed:
fix: resolved js error in course export (#34134)

syedsajjadkazmii committed:
feat: logout other sessions on email change (#33846)

* feat: logout other sessions on email change

* fix: updated the approach for session invalidation

* fix: update and add tests

* fix: update tests with descriptive comments

* feat: add integration tests

* fix: store email in session update

* fix: add setting for tests

* fix: fix tests

* feat: Upgrade Python dependency edx-drf-extensions (#34135)

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Co-authored-by: syedsajjadkazmii <syedsajjadkazmii@users.noreply.github.com>

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: syedsajjadkazmii <syedsajjadkazmii@users.noreply.github.com>

github-actions[bot] committed:
feat: Upgrade Python dependency ora2 (#34137)

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Co-authored-by: jansenk <jansenk@users.noreply.github.com>

deborahgu committed:
feat:  switching to a defined enum

feedback from code review

FIXES: APER-1322

deborahgu committed:
Merge branch 'master' into dkaplan1/APER-1322_certs-with-invalidation-records-arent-in-the-unavailable-status

deborahgu committed:
Merge pull request #34118 from openedx/dkaplan1/APER-1322_certs-with-invalidation-records-arent-in-the-unavailable-status

feat: data migration for some ancient legacy problems

AhtishamShahid committed:
fix: resolved cookie js bug on import page (#34139)

AhtishamShahid committed:
feat: added reported notification type (#34067)

saleem-latif committed:
feat: Upgrade Python dependency edx-enterprise

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

awais786 committed:
Merge branch 'master' into iamsobanjaved/django-42-lts

irfanuddinahmad committed:
Merge pull request #34130 from openedx/iahmad/ENT-8117

feat: Added XblockMixin for skill tagging

saleem-latif committed:
Merge pull request #34141 from openedx/saleem-latif/upgrade-edx-enterprise-039583f

feat: Upgrade Python dependency edx-enterprise

edx-requirements-bot committed:
chore: Updating Python Requirements (#34140)

deborahgu committed:
feat:  fixed one test

* fixed one test to accommodate a slightly modified workflow
* allowed the automated linter  to match current standards

Note:  as part of this process, I realized a lot of the tests in `programs/tests/test_tasks.py` are somewhat problematic. Some aren't real unit tests (allow calls to something other than the system under test), at least one for that reason it won't run on local at all, and some mock the wrong part of the system  or just don't match current flow.  I'm not modifying them as part of this, but they should be looked at.

FIXES: APER-3146

deborahgu committed:
Merge branch 'master' into dkaplan1/APER-3146_investigate-fix-exception-handling-in-program-cert-revocation

deborahgu committed:
feat: linting error

* unused import
* format

FIXES: APER-3146

Asespinel committed:
feat: added setting to disable the survey report banner entirely (#34092)

* feat: added setting to disable the survey report banner entirely

* fix: fixed unit test with new setting

* refactor: changed conditions for better code readability

* feat: added exception to stop the report from generating if the setting is set to false

* chore: updated the readme file to include the new setting

* refactor: move survey settings to common and disable admin by setting

* docs: typos in README

Co-authored-by: Tim McCormack <tmccormack@edx.org>

* refactor: set default values to survey report settings

* refactor: rename ENABLE_SURVEY_REPORT setting to SURVEY_REPORT_ENABLE

* test: fix quality tests

---------

Co-authored-by: Alejandro Cardenas <alecar.main@gmail.com>
Co-authored-by: Tim McCormack <tmccormack@edx.org>

deborahgu committed:
feat:  relinting

Trying with new autoformat settings.

FIXES: APER-3146

deborahgu committed:
Merge branch 'master' into dkaplan1/APER-3146_investigate-fix-exception-handling-in-program-cert-revocation

deborahgu committed:
feat: lint

autoformat errors

FIXES: APER-3146

arbrandes committed:
Merge pull request #34036 from raccoongang/ruzniaievdm/feat/unit-page-drf

feat: [FC-0044] Unit page API as DRF

Alec4r committed:
refactor: change survey report message location (#34126)

* refactor: change survey report message location

Alec4r committed:
feat: add state sent to state column (#34127)

deborahgu committed:
feat:  tweaks from code review

wording, switching to using a constant, fixing a string concat

FIXES: APER-3146

kaustavb12 committed:
fix: add missing function import in certificate template (#33904)

* fix: add missing function import in certificate template
* test: add test case to check certificates generated when GA4 is enabled

github-actions[bot] committed:
feat: Upgrade Python dependency edx-enterprise (#34144)

Added management command to fix `LearnerDataTransmissionAudit` table records.

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Co-authored-by: zamanafzal <zamanafzal@users.noreply.github.com>
Co-authored-by: Zaman Afzal <zamanafzal@gmail.com>

awais786 committed:
Merge branch 'master' into iamsobanjaved/django-42-lts

iamsobanjaved committed:
Merge pull request #34113 from openedx/iamsobanjaved/django-42-lts

chore!: upgrade Django version to 4.2 (LTS)

hamzawaleed01 committed:
feat: Upgrade Python dependency edx-enterprise

feat: update cornerstone client to store API calls in DB

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

hamzawaleed01 committed:
Merge pull request #34148 from openedx/hamzawaleed01/upgrade-edx-enterprise-cd8e2d2

feat: Upgrade Python dependency edx-enterprise

iamsobanjaved committed:
Revert "chore!: upgrade Django version to 4.2 (LTS)"

iamsobanjaved committed:
Merge pull request #34150 from openedx/revert-34113-iamsobanjaved/django-42-lts

Revert "chore!: upgrade Django version to 4.2 (LTS)"

navinkarkera committed:
feat: include custom relative dates flag info in outline api (#34058)

deborahgu committed:
Merge pull request #34129 from openedx/dkaplan1/APER-3146_investigate-fix-exception-handling-in-program-cert-revocation

feat: fix exception handling in program cert revocation

github-actions[bot] committed:
feat: Upgrade Python dependency ora2 (#34154)

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Co-authored-by: nsprenkle <nsprenkle@users.noreply.github.com>

Commit:
chore: geoip2: update maxmind geolite country database

Commit:
chore: update edx-enterprise dependencies version

ayesha-waris committed:
feat: added notifications when response is endorsed or answered (#34082)

* feat: added notification when response is endorsed or answered

* test: added and fixed test cases

* fix: fixed lint errors

* refactor: changed method name for readibility

* feat: added notification when my response is endorsed

* test: fixed failed test cases

iamsobanjaved committed:
feat!: upgrade Django version to 4.2 (LTS) (#34162)

* feat!: upgrade Django version to 4.2 (LTS)
---------

Co-authored-by: iamsobanjaved <iamsobanjaved@users.noreply.github.com>

github-actions[bot] committed:
feat: Upgrade Python dependency edx-enterprise (#34155)

Added logs and remove cornerstone from management command

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Co-authored-by: zamanafzal <zamanafzal@users.noreply.github.com>
Co-authored-by: Zaman Afzal <zamanafzal@gmail.com>

UsamaSadiq committed:
Revert "feat!: upgrade Django version to 4.2 (LTS) (#34162)" (#34165)

This reverts commit 1fc2e8a771a85db9b9a7e0b8c2da925bf81ed5bb.

Commit:
chore: rebase

katrinan029 committed:
Merge pull request #34159 from openedx/knguyen2/ent-8337

chore: bump edx-enterprise dependencies version

Commit:
Merge pull request #33922 from Zeit-Labs/plugins-i18n

feat: atlas pull plugins translation | FC-0012

Commit:
feat: add notification for course updates

iamsobanjaved committed:
feat!: upgrade Django version to 4.2 (LTS)

This reverts commit 23659d5ba84a4f5154c498d7ddacaf912caf7ac0.

bogdan3d committed:
fix: fixing the prompt view cancel button style for color and cursor shape (#33895)

hilltop16 committed:
Merge pull request #34153 from hilltop16/hilltop16/add-auth-JWKS-endpoint

feat: add auth JWKS endpoint

varshamenon4 committed:
Merge pull request #34021 from openedx/varshamenon4/refactor-cert-signals

refactor: move rejected exam event handlers

Commit:
fix: auth JWKS endpoint

The original implementation of the endpoint returned
a string instead of a JSON object. This fixes the
return value to be the JSON object.

hilltop16 committed:
Merge pull request #34176 from hilltop16/hilltop16/JWKS-endpoint-fixup

fix: return a JSON object in JWKS endpoint response

navinkarkera committed:
feat: add discussion settings to course index api (#34174)

viadanna committed:
fix: youtube race condition when GTM loads (#33649)

muhammadadeeltajamul committed:
feat: skip notification creation if context is not valid (#34133)

github-actions[bot] committed:
feat: Upgrade Python dependency edx-drf-extensions (#34189)

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Co-authored-by: syedsajjadkazmii <syedsajjadkazmii@users.noreply.github.com>

MueezKhan246 committed:
feat: Upgrade Python dependency edx-enterprise

update canvas client to store API calls in DB

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

MueezKhan246 committed:
Merge pull request #34190 from openedx/MueezKhan246/upgrade-edx-enterprise-a492b40

feat: Upgrade Python dependency edx-enterprise

edx-requirements-bot committed:
chore: Updating Python Requirements (#34183)

deborahgu committed:
update edx platform codeowners for aperture ownership (#34179)

* feat: tagging ownership of aperture on our owned apps

For all of the edx-platform apps for which 2u team aperture  is
responsible,  mark us as the owners.

FIXES: APER-3192

DmytroAlipov committed:
fix: ORA response with attached file (for master) (#33677)

The details are described in this discussion: https://discuss.openedx.org/t/ora-grading-returns-error/11482

timmc-edx committed:
feat: Support Django 4.2 in devstack (#34192)

With the Django 4.2 upgrade, devstack needs `CSRF_TRUSTED_ORIGINS` for MFEs
to work. This PR is probably not a complete list, but gets us started.

Ticket: https://github.com/openedx/edx-platform/issues/34180

nsprenkle committed:
feat: add ORA MFE to devstack CSRF trusted origins (#34197)

navinkarkera committed:
feat: filter courses by user language by default (#33647)

Adds a feature flag to filter courses by users preferred language by default

ayesha-waris committed:
feat: make notification channel headings clickable (#34194)

* feat: make notification channel headings clickable in notification preferences

* refactor: serializer code updated for better readability

* test: added a test for UserNotificationChannelPreferenceView API

* fix: updated the api test that was failing due to conflicts

---------

Co-authored-by: eemaanamir <eemaan.amir@gmail.com>

eemaanamir committed:
fix: survey xblock styling issues on mobile app

eemaanamir committed:
Merge branch 'openedx:master' into INF-1230

hamzawaleed01 committed:
feat: Upgrade Python dependency edx-enterprise

feat: register admin view for IntegratedChannelAPIRequestLogs

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

hamzawaleed01 committed:
Merge pull request #34205 from openedx/hamzawaleed01/upgrade-edx-enterprise-48f74fd

feat: Upgrade Python dependency edx-enterprise

nsprenkle committed:
feat: remove rollout percentage code for learner home (#34198)

This code allowed us to control rollout but is no longer needed.

deborahgu committed:
feat: transition learner_home to aperture (#34206)

previously assigned `learner_dashboard` to aperture (which is correct, aperture owns this) but the purpose of this effort was to assign `learner_home`, which is the backend people are referring to when they refer to "learner dashboard".

FIXES: APER-3174

justinhynes committed:
fix: fix default credentials internal service url in devstack settings

I was trying to test a communication pathway between the CMS and Credentials and requests were failing with the following error:
```
2024-02-07 19:25:46,524 ERROR 202 [celery.app.trace] [user 3] [ip 192.168.16.1] trace.py:270 - Task openedx.core.djangoapps.programs.tasks.update_credentials_course_certificate_configuration_available_date[898eae19-cd38-4857-a656-dc080d64a3c5] raised unexpected: ConnectionError(MaxRetryError("HTTPConnectionPool(host='localhost', port=18150): Max retries exceeded with url: /api/v2/course_certificates/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fc5b2c357c0>: Failed to establish a new connection: [Errno 111] Connection refused'))"))
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/urllib3/connection.py", line 174, in _new_conn
    conn = connection.create_connection(
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/urllib3/util/connection.py", line 95, in create_connection
    raise err
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/urllib3/util/connection.py", line 85, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused
```

I have updated the default value of the `CREDENTIALS_INTERNAL_SERVICE_URL` in our devstack settings to use the internal server URL and this fixed the issue.

justinhynes committed:
Merge pull request #34207 from openedx/jhynes/fix-devstack-credentials-url-cms

fix: fix default credentials internal service url in devstack settings

Commit:
chore: geoip2: update maxmind geolite country database

jerryankur committed:
fix: change deprecated defusedxml.lxml to defusedxml.common (#34177)

regisb committed:
fix!: infinite growth of cache when auto eviction is disabled

See discussion here: https://github.com/overhangio/tutor/pull/984

This is a breaking change that will explicitely set the timeout of
course structure cache entries to 1 week, instead of being unlimited. If
you wish to revert to the former behaviour, you should set
`course_structure_cache["TIMEOUT"] = None`.

The course structure cache keys were previously set with an explicit
timeout of `None` which was overriding the cache default timeout of 2h.
This was OK in environments where the cache is configured with a maximum
memory limit and an automatic key eviction strategy. But in others (such
as Tutor), the course structure cache could grow infinitely.

It was agreed that course structure cache keys should be long-lived but
should respect the default cache structure timeout. Thus, we set here
the TTL to 1 week.

We can also configure Tutor to use a cache eviction policy. But that
means we need to set a `maxmemory` value in Redis. It's not possible to
set a value that will be appropriate for everyone:
- if it's higher than the total memory (e.g: in small instances), server
  will crash before the cache is filled.
- if it's too low (e.g: in large instances), the whole platform will abruptly
  slow down as many cache entries are suddenly evicted.

That question of whether Tutor should define a cache eviction policy is
still under discussion, but it can be resolved independently of this
change.

justinhynes committed:
docs: add diagrams visualizing "available date" and "visiblity date"

A PR that adds a diagram that visualizes how an update to a course run's availability date propogates through our systems.

ChrisChV committed:
feat: Export ID on Taxonomy (in Studio's tagging API) (#34143)

justinhynes committed:
Merge pull request #34211 from openedx/jhynes/APER-3153_cad-spike

docs: add diagrams visualizing "available date" and "visiblity date"

rpenido committed:
refactor: remove ContentObjectTag model and related functions (#34146)

sundasnoreen12 committed:
Merge pull request #34203 from eemaanamir/INF-1230

fix: survey xblock styling issues on mobile app

muhammadadeeltajamul committed:
feat: allow staff and admin to pin posts (#34169)

justEhmadSaeed committed:
feat: Upgrade Python dependency edx-enterprise

[ENT-8008] feat: add unique constraint on learner transmission audit

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

justEhmadSaeed committed:
Merge pull request #34217 from openedx/justEhmadSaeed/upgrade-edx-enterprise-b28db57

feat: Upgrade Python dependency edx-enterprise

christopappas committed:
fix: require email verification on FA form (#34219)

fix: broken test lms/djangoapps/courseware/tests/test_views.py

fix: refactor test

christopappas committed:
fix: give FA form user friendly error message (#34222)

christopappas committed:
chore: add logging to fa form (#34224)

* chore: add logging to fa form

* chore: formatting

edx-requirements-bot committed:
chore: Updating Python Requirements (#34225)

zainab-amir committed:
feat: update authn mfe devstack settings (#34221)

salman2013 committed:
Update all edx-platform REST endpoints to support JWT Auth (#34152)

* chore: update API endpoints to support default JWT auth

The default DRF Auth classes were recently updated to allow for both JWT and Session auth by default. Any endpoint that overrides the AUTHENTICATION_CLASSES but has just session, just JWT or just both of those should be updated to remove the override.

Details in https://github.com/openedx/edx-platform/issues/33662

zacharis278 committed:
feat: send course role events to the event bus (#34158)

Notify the event bus when a user's role in a course is added or removed

ahtesham-quraish committed:
fix: update registration api response and replace username with name (#34226)

Update api registration api response and add full_name in response

lewisemm committed:
chore: Use preferred collection method `delete_one` (#34214)

`remove` method of Collection objects has been deprecated in favour
of either `delete_one` or `delete_many`.

This change will address 36 deprecation warnings that are generated from
test runs.

Co-authored-by: Lewis Kabui <lewisemm@users.noreply.github.com>

MueezKhan246 committed:
feat: Upgrade Python dependency edx-enterprise

recorded degreed, moodle, and SAP API calls

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

MueezKhan246 committed:
Merge pull request #34232 from openedx/MueezKhan246/upgrade-edx-enterprise-4ee11d3

feat: Upgrade Python dependency edx-enterprise

Commit:
refactor: remove the managment command populate user emails on braze (#34228)

salman2013 committed:
Remove bok-choy reference (#34216)

* chore: remove bok-choy references

Ian2012 committed:
feat: add open_managed team type (#33672)

Ref: https://openedx.atlassian.net/wiki/spaces/COMM/pages/3885760525/Open+Managed+Group+Type

alex-sheehan-edx committed:
feat: Upgrade Python dependency edx-enterprise

enterprise version bump 4.11.12

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

yusuf-musleh committed:
feat: Serialize tag data in OLX for blocks (#34145)

alex-sheehan-edx committed:
Merge pull request #34235 from openedx/alex-sheehan-edx/upgrade-edx-enterprise-5a36fa9

feat: Upgrade Python dependency edx-enterprise

Commit:
feat: add discussions MFE to CSRF Trusted origin list for devstack

AhtishamShahid committed:
fix: added preference visibility to reported notification (#34188)

BryanttV committed:
feat: expose `hide_from_toc` field for course blocks in outline API (#33955)

These changes are part of the effort made to implement https://openedx.atlassian.net/wiki/spaces/OEPM/pages/3853975595/Feature+Enhancement+Proposal+Hide+Sections+from+course+outline

Alec4r committed:
feat: add dismiss button to survey report banner (#34160)

* feat: add dismiss button to survey report banner

* refactor: move banner script to js file

* fix: remove not in conditional

ayesha-waris committed:
feat: modified author labels to add moderator (#34239)

Co-authored-by: sohailfatima <23100065@lums.edu.pk>

AhtishamShahid committed:
fix: updated failing notification pref unit test (#34243)

github-actions[bot] committed:
feat: Upgrade Python dependency ora2 (#34244)

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Co-authored-by: jansenk <jansenk@users.noreply.github.com>
Co-authored-by: Jansen Kantor <jkantor@edx.org>

michaelroytman committed:
temp: add supplemental logging to debug IDV issues (#34248)

This commit adds some supplemental, more verbose logging to the results_callback view in the verify_student Djangoapp. This endpoint is called by identity verification (IDV) providers to POST an IDV review to edX.

We are experiencing issues with receiving IDV reviews from our IDV provider, and these logs will help us diagnose whether there is an issue in edX's systems.

These logs will be removed after our investigation is complete.

rpenido committed:
feat: export all course tags as csv (#34091)

christopappas committed:
feat: make FA form error messaging more descript (#34247)

* feat: make FA form error messaging more descript

* chore: quality

* fix: split up tests

* fix: make test more specific

* chore: fix comment typo

* chore: fix comment typo

pomegranited committed:
perf: reduce number of queries for content tagging endpoints (#34200)

jawad-khan committed:
feat: Add course price in mobile enrollment api (#34255)

* feat: Add course price in mobile enrollment api

OmarIthawi committed:
chore: pull_translations: use less Makefile commands and move it to Python

`mkdir` should be in Python

mubbsharanwar committed:
fix: update activation email (#34136)

update account activation email message copy in body

VAN-1800

Commit:
refactor: Remove unused code for merchandising 2U LOBs (#34230)

MueezKhan246 committed:
feat: Upgrade Python dependency edx-enterprise

Added channel_name for api call logs records

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

edx-requirements-bot committed:
chore: Updating Python Requirements (#34258)

MueezKhan246 committed:
Merge pull request #34262 from openedx/MueezKhan246/upgrade-edx-enterprise-3a05fd2

feat: Upgrade Python dependency edx-enterprise

alangsto committed:
feat: replace la waffle flag with django setting (#34236)

Commit:
Merge pull request #34257 from Zeit-Labs/pythonize_pull_translations

chore: pull_translations: use less Makefile commands and move it to Python | FC-0012

hamzawaleed01 committed:
feat: Upgrade Python dependency edx-enterprise

feat: unlink canvas user if decommissioned on canvas side

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

hamzawaleed01 committed:
Merge pull request #34268 from openedx/hamzawaleed01/upgrade-edx-enterprise-92cde34

feat: Upgrade Python dependency edx-enterprise

mariajgrimaldi committed:
fix: use non_atomic_requests decorator in handle_block view (#34020)

mariajgrimaldi committed:
refactor: address PR reviews

mariajgrimaldi committed:
refactor: address PR reviews

awais-ansari committed:
feat: add discussion configuration redirect in unit settings (#34266)

* feat: add discussion configuration redirect in unit settings

* test: fix test course index file test cases after adding discussion redirection

BryanttV committed:
feat: update `initialize` endpoint and create `assessments/feedback` endpoint in ORA Staff Grader (#33632)

farhan committed:
Move user retirement scripts code from the tubular repo (#34063)

* refactor: Migragte user retirement scripts code from the tubular repo

ormsbee committed:
feat!: Switch v2 libraries to Learning Core data models (#34066)

This moves the Content Libraries V2 backend from Blockstore [1] over to
Learning Core [2] For high-level overview and rationale of this move, see
the Blockstore DEPR [3]. There are several follow-up tasks [4], most notably
adding support for static assets in libraries.

BREAKING CHANGE: Existing V2 libraries, backed by Blockstore, will stop
working. They will continue to be listed in Studio, but their content
will be unavailable. They need to be deleted (via Django admin) or manually
migrated to Learning Core. We do not expect production sites to be in
this situation, as the feature has never left "experimental" status.

[1] https://github.com/openedx-unsupported/blockstore
[2] https://github.com/openedx/openedx-learning/
[3] https://github.com/openedx/public-engineering/issues/238
[4] https://github.com/openedx/edx-platform/issues/34283

sundasnoreen12 committed:
fix: added table tags in allowed tags

timmc-edx committed:
fix: Support running pull_translations with base dependencies (#34285)

Since `make pull_translations` is now performed at build time (in the new
Atlas workflow), it will generally only have access to base dependencies.
However, it depends on the `i18n_tools` command that was only installed at
the testing layer. This moves the dependency from testing to base.

(There is no increase in artifact size, as ora2 already depends on it and
is installed at the base later.)

This also adds `API_ACCESS_MANAGER_EMAIL` to the minimal settings
so that `make pull_translations` can be run using production envs in a
local environment (to test the lack of dependency on developer-level deps.)

This supports OEP-58.

Commit:
chore: version bump

katrinan029 committed:
Merge pull request #34287 from openedx/knguyen2/version-bump

chore: version bump

sundasnoreen12 committed:
Merge pull request #34286 from openedx/sundas/INF-1148

fix: added table tags in allowed tags

alangsto committed:
feat: remove learning assistant waffle flag (#34272)

jansenk committed:
feat: add db models for course reset feature (#34282)

* feat: add db models for course reset feature

* style: quality

* fix: read only fields when creating / updating model

ahtesham-quraish committed:
fix: replace username with name in contact us form (#34252)

Description: Replace username with name in contact us form
VAN-1822

ayesha-waris committed:
fix: Error message does not show  when  user clicks Dismiss button (#34277)

MueezKhan246 committed:
feat: Upgrade Python dependency edx-enterprise

added management command to clear out excessive records for API log table

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

MueezKhan246 committed:
Merge pull request #34293 from openedx/MueezKhan246/upgrade-edx-enterprise-c61df90

feat: Upgrade Python dependency edx-enterprise

timmc-edx committed:
chore: Update org to openedx for move-affected repos (#34297)

See https://github.com/edx/edx-arch-experiments/issues/558

awais-ansari committed:
fix: newline and alignment bug in survey option mobile view (#34295)

zainab-amir committed:
feat: update account verification email context (#34294)

moeez96 committed:
refactor: Increase size for mobile config singelton value (#34288)

Co-authored-by: Abdul  Moeez Zahid <abdul.moeez@025907584957.2tor.net>

arbrandes committed:
Merge pull request #31976 from openedx/MJG/acc-edx-brand-removal

feat: Remove edX branding on account deletion

alex-sheehan-edx committed:
feat: Upgrade Python dependency edx-enterprise

edx enterprise version bump 4.12.4

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

alex-sheehan-edx committed:
Merge pull request #34301 from openedx/alex-sheehan-edx/upgrade-edx-enterprise-d95ac60

feat: Upgrade Python dependency edx-enterprise

jansenk committed:
feat: add list endpoint for course reset enabled enrollments

Danyal-Faheem committed:
fix: hide help button in unauthenticated studio header (#34249)

Co-authored-by: Zia Fazal <zia.fazal@arbisoft.com>

Anas12091101 committed:
fix: fixed asides only view

hajorg committed:
feat: add post endpoint for course reset

hajorg committed:
fix: resolve lint issues

github-actions[bot] committed:
feat: Upgrade Python dependency edx-enterprise (#34304)

feat: Adding a group membership to the EnterpriseCustomerUserReadOnlySerializer

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Co-authored-by: kiram15 <kiram15@users.noreply.github.com>
Co-authored-by: Kira Miller <31229189+kiram15@users.noreply.github.com>

OmarIthawi committed:
feat: fix pull_translations for production deployment

Makefile paver usage replaced with manage.py.
This avoids running the production-unsuitable
`pavelib.prereqs.install_prereqs` step during deployments.

hajorg committed:
Merge pull request #34303 from openedx/jkantor/list-available

feat: add list endpoint for course reset enabled enrollments

ahtesham-quraish committed:
fix: add vanguards to CODEOWNERS (#34291)

Add vanguards to CODEOWNERS
VAN-1817

ayesha-waris committed:
feat: updated api to get all question type reponses (#34215)

* feat: updated api to get all question type reponses

* test: fixed and added new test cases

github-actions[bot] committed:
feat: Upgrade Python dependency edx-enterprise (#34312)

fix: Proximus learner transmission data failures

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Co-authored-by: sameenfatima78 <sameenfatima78@users.noreply.github.com>

mariajgrimaldi committed:
feat: make hide from TOC a visibility section setting (#33952)

Exposes the hide_from_toc xblock attribute so course authors can configure it as a section visibility option in Studio. Before this change, the Hide from TOC functionality was mainly used by OLX components. Hence, it wasn't available for configuration through the Studio UI. Still, its implementation existed in the platform and could be used by setting the attribute: hide_from_toc=true as part of the OLX definition.
Ref: https://openedx.atlassian.net/wiki/spaces/OEPM/pages/3853975595/Feature+Enhancement+Proposal+Hide+Sections+from+course+outline

github-actions[bot] committed:
feat: Upgrade Python dependency edx-enterprise (#34314)

* feat: Upgrade Python dependency edx-enterprise

Bump version of edx-enterprise to v4.13.0.

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

---------

Co-authored-by: adamstankiewicz <adamstankiewicz@users.noreply.github.com>
Co-authored-by: Adam Stankiewicz <agstanki@gmail.com>

Commit:
Merge pull request #33690 from Inferato/lytvynenko/bulk_email_checkbox_master

fix: course mode added to the metadata

mubbsharanwar committed:
fix: fir segment event (#34279)

fire segment event for PWNED_PASSWORD on registration page password validation

VAN-1830

ahtesham-quraish committed:
fix: add country field error message (#34316)

Description: Add country field error message in api response
VAN-1862

Commit:
Merge pull request #34306 from Zeit-Labs/compile_js_no_paver

feat: fix pull_translations for production deployment

ormsbee committed:
docs: add pruning-related warning messages in MongoDB connection

We migrated the source of truth for what the active draft and published
versions of course and v1 library content are to the
SplitModulestoreCourseIndex Django model. But the contentpruning
code (structures.py) that was developed in tubular and will be moved to
edx-platform is not aware of this newer model, and still only pulls its
source of truth from MongoDB. So we *must* continue to do writes to
MongoDB, or the pruning code will start pruning live versions.

The longer term fix for this is to make the pruning code aware of
SplitModulestoreCourseIndex, which will be easier once it's moved into
edx-platform.

Commit:
chore: edx-enterprise version bump

katrinan029 committed:
Merge pull request #34319 from openedx/knguyen2/ent-8531

chore: edx-enterprise version bump

hajorg committed:
Merge pull request #34305 from openedx/hajorg/au-1847-create-course-reset

feat: add post endpoint for course reset

rpenido committed:
feat: export tagged content library as CSV (#34246)

jristau1984 committed:
chore: update arch bom team name

Commit:
fix: add notification expiry settings to cms

jristau1984 committed:
Merge pull request #34326 from openedx/catalog-info-updates

chore: update arch bom team name

Commit:
fix: fix content_url for course update notification

alex-sheehan-edx committed:
feat: Upgrade Python dependency edx-enterprise

enterprise version bump 4.13.3

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

OmarIthawi committed:
feat!: atlas pull for studio-frontend translations (#34261)

This pull request pulls translations via atlas for `studio-frontend` and refactor `load_sfe_i18n_messages` to load new translations into `conf/plugins-locale/studio-frontend` instead of relying on deprecated node.js package bundled translations.

alex-sheehan-edx committed:
Merge pull request #34333 from openedx/alex-sheehan-edx/upgrade-edx-enterprise-e3467f9

feat: Upgrade Python dependency edx-enterprise

tspring5000 committed:
fix: Open Responses Styling and Layout (#33717)

* fix: table styling on Instructor Dashboard

Ref: https://github.com/openedx/wg-build-test-release/issues/223
The Open Responses tab on instructor dashboard on LMS has a few issues with table styling; there isn't enough spacing between some columns which makes the text difficult to read, and the bottom of some numbers in the summary table are cutoff
This change adds padding to the right of each column to ensure longer text is still legible and changes the line height from 1em to 1 (unitless), which looks the same but doesn't cut off the bottom

* fix: remove whitespace
Remove extra whitespace line 1862 for better formatting

Anas12091101 committed:
fix: pylint import errors

ayesha-waris committed:
fix: update API endpoint to return task due time for future scheduled tasks (#34332)

* fix: update API endpoint to return task due time for future scheduled tasks

* test: updated InstructorEmailContentList tests to accomodate changes

* fix: returend unformatted created date to support easy conversion at frontend

* test: updated tests to accommodate unformatted date value in API response

* refactor: removed duplicate code and use of random numbers

---------

Co-authored-by: eemaanamir <eemaan.amir@gmail.com>

michaelroytman committed:
feat: add new management command to approve submitted ID verification attempts

This pull requests adds a new management command approve_id_verifications to manually approve submitted ID verification attempts (i.e. instances of the SoftwareSecurePhotoVerifications model).

BryanttV committed:
feat: add share link button when hide from toc is enabled in sections (#34043)

* feat: add share link modal when hide from toc is enabled

Adds a new button in the child subsections of sections with Hide From TOC enabled.
This button displays a new modal with two tabs. The first tab displays a button
that allows you to copy the link of that subsection to the clipboard. The second
tab displays a button that allows you to copy the embedded link of the same
subsection to the clipboard.

Ref: https://openedx.atlassian.net/wiki/spaces/OEPM/pages/3853975595/Feature+Enhancement+Proposal+Hide+Sections+from+course+outline

BryanttV committed:
feat: show message in subsection visibility tab when hft is enabled (#33981)

Shows a message in the visibility tab of the subsections when the Hide From TOC is enabled.

Ref: https://openedx.atlassian.net/wiki/spaces/OEPM/pages/3853975595/Feature+Enhancement+Proposal+Hide+Sections+from+course+outline

rayzhou-bit committed:
fix: show answer should show correct answers (#34335)

DanielVZ96 committed:
build: add MongoDB 7 to the CI (#34213)

xitij2000 committed:
feat: add support for zip files to course and library import code (#34191)

Co-authored-by: Rodrigo Ferreira de Souza <rodfersou@gmail.com>

michaelroytman committed:
Merge pull request #34340 from openedx/mroytman/COSMO-210-idv-approval-management-command

Add new management command to approve submitted ID verification attempts.

rpenido committed:
feat: paste tags when pasting xblocks with tag data (#34270)

irfanuddinahmad committed:
feat: Upgrade Python dependency edx-enterprise

Improved Enterprise Customer search on the django admin page for EnterpriseCustomerCatalog

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Commit:
fix: removed waffle switch around oep-50 filter

irfanuddinahmad committed:
Merge pull request #34346 from openedx/irfanuddinahmad/upgrade-edx-enterprise-cb6801d

feat: Upgrade Python dependency edx-enterprise

irfanuddinahmad committed:
Merge pull request #34347 from openedx/iahmad/ENT-7614-2

fix: removed waffle switch around oep-50 filter

hamzawaleed01 committed:
feat: add frontend-app-admin-portal to whitelisted origins

hamzawaleed01 committed:
Merge pull request #34354 from openedx/hamzawaleed01/devstack-whitelist-admin-portal

feat: add frontend-app-admin-portal to whitelisted origins

farhan committed:
chore: Moved structures.py from tubular repository (#34328)

* chore: Moved structures.py from tubular repository

ruzniaievdm committed:
feat: add sequence_ids to container response (#34337)

* feat: add sequence_ids to container response

* fix: PR comments fix

* fix: update

---------

Co-authored-by: monteri <36768631+monteri@users.noreply.github.com>

ruzniaievdm committed:
feat: [FC-0044] XBlock's children API as DRF (#34055)

* feat: XBlock's children API as DRF

* fix: 500 error appears if user adds a Content Experiment

* fix: wrap into try/except block getting icon for xblock (#2509)

* fix: wrap into try/except block getting icon for xblock

* fix: revision after review

mariajgrimaldi committed:
feat: return navigation disabled as sequence metadata (#34049)

Return navigation disabled as sequence metadata when Hide From TOC
is enabled, so the student cannot navigate to another sequences in
the course outline.
https://openedx.atlassian.net/wiki/spaces/OEPM/pages/3853975595/Feature+Enhancement+Proposal+Hide+Sections+from+course+outline

asadali145 committed:
fix: clear block structure cache when usage key is not found (#33963)

hamzawaleed01 committed:
feat: Upgrade Python dependency edx-enterprise

feat: disable failing content transmissions for 24hrs

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

hamzawaleed01 committed:
Merge pull request #34359 from openedx/hamzawaleed01/upgrade-edx-enterprise-19c04fc

feat: Upgrade Python dependency edx-enterprise

MueezKhan246 committed:
feat: Upgrade Python dependency edx-enterprise

added get_queryset for fix of integrated channel api logs loading

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

KristinAoki committed:
feat: update page renders for studio-frontend pages (#34329)

* feat: update page renders for studio-frontend pages

* fix: checklist redirect url

* fix: failing tests

* fix: checklist link

* fix: redirect url typo

* fix: no newline error

MueezKhan246 committed:
Merge pull request #34362 from openedx/MueezKhan246/upgrade-edx-enterprise-ec1e58f

feat: Upgrade Python dependency edx-enterprise

feanil committed:
docs: Update README with new Mongo Version

Support was verified via https://github.com/openedx/edx-platform/pull/34213
but we forgot to update the README with this info.

Commit:
Merge pull request #34365 from openedx/feanil/edx-platform-mongo

docs: Update README with new Mongo Version

hamzawaleed01 committed:
feat: Upgrade Python dependency edx-enterprise

fix: add missing filter to disable failing transmissions for 24hrs

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

hamzawaleed01 committed:
Merge pull request #34371 from openedx/hamzawaleed01/upgrade-edx-enterprise-42f7fef

feat: Upgrade Python dependency edx-enterprise

OmarIthawi committed:
feat!: remove Transifex calls for FC-0012 - OEP-58 (#34355)

OmarIthawi committed:
fix: avoid removing Open_edX.egg-info/ with `git clean` (#34380)

git clean when used with glob pattern (*/) removes non-translations
files.

using `$ rm` to fix the issue.

Anas12091101 committed:
fix: empty commit to run the tests
